policyenforcer.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratify
+
+import (
+	"context"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// registeredPolicyEnforcers saves the registered policy enforcer factories.
+var registeredPolicyEnforcers map[string]func(config CreatePolicyEnforcerOption) (PolicyEnforcer, error)
+
+// ReferrerValidationReport describes the results of verifying an associated artifact
+// and its nested artifacts by available verifiers.
+type ReferrerValidationReport struct {
+	// Subject is the subject artifact of the current referrer being verified. Required.
+	Subject string
```

Comment:
What exactly is a `Subject`? is it a reference string of subject?

Reply:
The name of `Subject` is somehow ambiguous. Should it just be `Artifact` or `Target`?